sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs

```diff
@@ -2604,6 +2605,47 @@ Response<Blobs.Models.BlobDownloadInfo> response
         #endregion Read To
 
         #region Upload
+        /// <summary>
+        /// The <see cref="Upload(Stream, DataLakeFileUploadOptions, CancellationToken)"/>
+        /// operation creates and uploads content to a file.  If the file already exists, its content will be overwriten.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update" />.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="Stream"/> containing the content to upload.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// state of the updated file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<PathInfo> Upload(
+            Stream content,
+            DataLakeFileUploadOptions options,
```

Comment:
should options be optional?

Reply:
No, because if options is optional, there will be ambiguity between 
```Upload(Stream content, DataLakeFileUploadOptions options = default, CancellationToken cancellationToken = default)```
and
```UploadAsync(Stream content, PathHttpHeaders httpHeaders = default)```,
 breaking existing customer code.

Reply:
Should this be called options then ?:-) I guess we'll drop conflicting overload in V13 and then make it optional.

Reply:
What happens if someone is trying to pass `null` as second parameter. Would this be ambigous?

Reply:
We're supposed to use the Options suffix for optional parameter groupings.

If someone passed in a (stream, default, default), it would conflict with ```Upload(Stream content, bool overwrite = false, CancellationToken cancellationToken = default)```.

Either way, I'm pretty sure options has to be required or it will break the case where the user passing in just a stream.